### DISCUSSION TITLE
Compare methods source code

### DISCUSSION THREAD

**Discussion by Mirocos** [ImageMagick/ImageMagick#4013] [Help] [answered]
Hi, I'm new to ImageMagick, I want to learn how to realize "compare" tool.I found some file called "compare.c" and "compare.h" in directory "MagicWand". Are these all I need to learn?
By the way, any possible to build this by CMake?

**Comment by fmw42** [comment]
Why do you want the source code?  You should just experiment with the command line until you are more familiar after installing ImageMagick from binary. What is your platform?

**Reply by Mirocos** [reply]
Ubuntu18.04LTS, Actually I want to compare two textures dumped from my OpenGL program. Not just the "compare" command line.

**Comment by snibgo** [comment, marked as answer]
Knowledge of the source code isn't needed to use the software. But you can if you want, of course.

MagickWand/compare.c is merely the frontend. It processes the command arguments, reads the inputs, and writes the output.

The real work is done in MagickCore/compare.c.

Some people build IM with CMake. I use the GNU toolset.

**Reply by Mirocos** [reply]
Really helpful, I found out the way to use IM compare feature in my program with CMake.

**Reply by Mirocos** [reply]
sorry, I have a another question: when I link static library to my program, compiler will say undefined reference in *.a. But if I link dynamic library, it works, what make this happend?

**Comment by snibgo** [comment]
I don't know, sorry. It may be some option in CMake.
